meteor-slingshot/services/cloudflare-r2.js

```diff
@@ -0,0 +1,179 @@
+const crypto = Npm.require('crypto');
+const url = Npm.require('url');
+
+Slingshot.CloudflareR2 = {
+  accessId: 'AccessKeyId',
+  secretKey: 'SecretAccessKey',
+
+  directiveMatch: {
+    bucket: String,
+    accountId: String,
+    endpoint: String,
+    cdn: String,
+    region: Match.Optional(String),
+    AccessKeyId: String,
+    SecretAccessKey: String,
+    key: Match.OneOf(String, Function),
+    expire: Match.Where(function (expire) {
+      check(expire, Number);
+      return expire > 0;
+    }),
+  },
+
+  directiveDefault: {
+    region: 'auto',
+    expire: 5 * 60 * 1000,
+  },
+
+  getContentDisposition: async function (method, directive, file, meta) {
+    var getContentDisposition = directive.contentDisposition;
+
+    if (!_.isFunction(getContentDisposition)) {
+      getContentDisposition = function () {
+        var filename = file.name && encodeURIComponent(file.name);
+
+        return (
+          directive.contentDisposition ||
+          (filename &&
+            'inline; filename="' +
+              filename +
+              "\"; filename*=utf-8''" +
+              filename)
+        );
+      };
+    }
+
+    return getContentDisposition.call(method, file, meta);
+  },
+
+  upload: async function (method, directive, file, meta) {
+    try {
+      const policy = new Slingshot.StoragePolicy()
+        .expireIn(directive.expire)
+        .contentLength(0, Math.min(file.size, directive.maxSize || Infinity));
+
+      let key;
+      if (_.isFunction(directive.key)) {
+        key = await Promise.resolve(directive.key.call(method, file, meta));
```

Comment:
Why the need for `Promise.resolve`? If the `key` function is not `async`, `await` will have no effect

Reply:
Also, you could use a ternary here instead, would be cleaner to read.

Reply:
You're right. Fixed.